src/local.rs

```diff
@@ -641,6 +641,7 @@ impl LocalFileSystem {
         let walkdir = WalkDir::new(root_path)
             // Don't include the root directory itself
             .min_depth(1)
+            .sort_by_file_name()
```

Comment:
This will list every file in the directory, sort it, and then return it - at the very least this should be explicitly opt-in as directories can get very large - the standard library APIs are intentionally lazy

Reply:
That's true and you don't always need it. But sometimes you absolutely do; offset listing just makes no sense without sorting.

- offset listing: need sorting
- iterator listing: don't care about sorting, higher perf. without sorting

opt-out could be a good option too given those requirements